katdal/visdatav4.py

```diff
@@ -277,24 +279,42 @@ def __init__(self, source, ref_ant='', time_offset=0.0, applycal='',
         self.sensor['Observation/label'] = label
         self.sensor['Observation/compscan_index'] = CategoricalData(list(range(len(label))),
                                                                     label.events)
-        # Use the target sensor of reference antenna to set target for each scan
-        target = self.sensor.get(self.ref_ant + '_target')
-        # Remove initial blank target (typically because antenna starts stopped)
-        if len(target) > 1 and target[0] == 'Nothing, special':
-            target.events, target.indices = target.events[1:], target.indices[1:]
-            target.events[0] = 0
+        # Use the relevant target sensor to set target for each scan
+        ref_target = self.ref_ant or 'cbf'
+        target = self.sensor.get(ref_target + '_target')
         # Move target events onto the nearest scan start
         # ASSUMPTION: Number of targets <= number of scans
         # (i.e. only a single target allowed per scan)
         target.align(scan.events)
+        # Remove initial target if antennas start in mode STOP
+        # (typically left over from previous capture block)
+        for segment, scan_state in scan.segments():
+            # Keep going until first non-STOP scan or a new target is set
+            if scan_state == 'stop' and target[segment.start] is target[0]:
+                continue
+            # Only remove initial target event if we move to a different target
+            if target[segment.start] is not target[0]:
+                target.events = target.events[1:]
+                target.indices = target.indices[1:]
```

Comment:
I'm discarding *target* events, not *scan* events. Since targets are extracted with `allow_repeats=False` (another assumption that I could verify...), there will be one event for `target[0]` and another for the next unique target. Only the first one needs to be removed and then the sensor is normalised.

Reply:
Ensure that only one event has to be removed by calling `target.remove_repeats()`.